WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormRemoteActionUseCase.swift

```diff
@@ -24,28 +25,80 @@ final class ProductFormRemoteActionUseCase {
     ///   - onCompletion: Called when the remote process finishes.
     func addProduct(product: EditableProductModel,
                     password: String?,
+                    successEventName: WooAnalyticsStat = .addProductSuccess,
+                    failureEventName: WooAnalyticsStat = .addProductFailed,
                     onCompletion: @escaping AddProductCompletion) {
         addProductRemotely(product: product) { productResult in
             switch productResult {
             case .failure(let error):
-                ServiceLocator.analytics.track(.addProductFailed, withError: error)
+                ServiceLocator.analytics.track(successEventName, withError: error)
                 onCompletion(.failure(error))
             case .success(let product):
                 // `self` is retained because the use case is not usually strongly held.
                 self.updatePasswordRemotely(product: product, password: password) { passwordResult in
                     switch passwordResult {
                     case .failure(let error):
-                        ServiceLocator.analytics.track(.addProductFailed, withError: error)
+                        ServiceLocator.analytics.track(failureEventName, withError: error)
                         onCompletion(.failure(.passwordCannotBeUpdated))
                     case .success(let password):
-                        ServiceLocator.analytics.track(.addProductSuccess)
+                        ServiceLocator.analytics.track(successEventName)
                         onCompletion(.success(ResultData(product: product, password: password)))
                     }
                 }
             }
         }
     }
 
+    /// Adds a copy of the input product remotely.
+    /// - Parameters:
+    ///   - originalProduct: The product to be duplicated remotely.
+    ///   - onCompletion: Called when the remote process finishes.
+    func duplicateProduct(originalProduct: EditableProductModel,
+                          password: String?,
+                          onCompletion: @escaping DuplicateProductCompletion) {
+        let productModelToSave: EditableProductModel = {
```

Comment:
I think it's usually good to declare type explicitly, especially for complicated computed variables to avoid long compiling time for type inference. I removed this and it seems that CI (Xcode 13) [fails](https://buildkite.com/automattic/woocommerce-ios/builds/7970#018353f5-ca4e-4823-8d7a-6673769d1a4c) to infer the type, so I restore it back 😅.